integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAlterTableCompactionCommand.scala

```diff
@@ -156,10 +158,19 @@ case class CarbonAlterTableCompactionCommand(
       )
       .equalsIgnoreCase("true")
 
-    // if system level compaction is enabled then only one compaction can run in the system
-    // if any other request comes at this time then it will create a compaction request file.
-    // so that this will be taken up by the compaction process which is executing.
-    if (!isConcurrentCompactionAllowed) {
+    // If the carbon table is a child datamap then start compaction for the same using Load
```

Comment:
Please add an interface at `DataManagementFunc` and give 2 implementations , one for maintable and another one for agg table. Execute compaction depends on table type